.github/workflows/mkdocs.yml

```diff
@@ -17,6 +17,13 @@ jobs:
       - name: 📥 Checkout repository
         uses: actions/checkout@v3
 
+      - name: 🔨 Install rzk proof assistant
+        uses: jaxxstorm/action-install-gh-release@v1.10.0
+        with:
+          repo: fizruk/rzk
+          tag: v0.4.1
```

Comment:
Since we are in the `rzk` repository, why not use the version built for Haddock? Why fix the version here?

Reply:
And how do I do that? Add an [upload-artifact](https://github.com/actions/upload-artifact) step in the Haddock Action and download it here?

Reply:
Yes, exactly, thank you!

Reply:
It turns out that uploading and downloading artifacts can only be done between different jobs in the same workflow. I removed the tag so it would take the latest release for now, but I would refer to [this workflow](https://github.com/illright/attractions/blob/main/.github/workflows/deploy-pr-docs.yml) as a workaround for sharing artifacts between different workflows (though it doesn't apply exactly to the currently defines workflows, so some refactoring to the CI/CD process is needed)

Reply:
Perhaps, we should merge workflows then?

Reply:
This is a possibility, but they both have different triggers so it wouldn't be ideal. I think the workflows can benefit from some refactoring overall, but I'm wondering if it's more effort than it's worth (especially at the time being).
In short, the current options are:
1. Refactor all the workflows to optimize the logic and avoid redundant builds (inspired by [these workflows](https://github.com/illright/attractions/tree/main/.github/workflows) I coauthored before)
2. Merge these 2 specific jobs into one workflow, overlooking the difference in triggers (or perhaps adding some `if:`s
3. Just keep it as it is for now, fetching the executable from the latest release

Which one do you prefer?

Reply:
Let's keep it as is and refactor workflows later (create an issue). Thank you!